### PR TITLE
handled invalid AIS coordinates

### PR DESCRIPTION
**Pull request by jamenkaye**
`AISCallback()` in Sailbot.py now removes any AIS boats with latitude outside [-90, 90] or longitude outside [-180, 180].

- Imade a utilities function `checkLatLonIsValid()` to check if a latlon is valid. Is this useful or should I delete this function and just check for validity in `AISCallback()`?
- I wanted to use the `List.remove()` function but then the loop seemed to run the incorrect number of times if I removed items from the list of ships during the loop. I don't really like that I needed to create a whole new list of AIS boats though, as this seems inefficient. Is there a better way I should do this? For example I could `pop()` the element at `[(loop index) - (number of boats removed so far)]
- Should I make it so that we can easily toggle between bounding invalid coordinates to valid equivalents vs deleting AIS boats with invalid coordinates?

### CODE REVIEW

**Review by patrick-5546 — CHANGES_REQUESTED** [review]
> I wanted to use the List.remove() function but then the loop seemed to run the incorrect number of times if I removed items from the list of ships during the loop. I don't really like that I needed to create a whole new list of AIS boats though, as this seems inefficient. Is there a better way I should do this?

I think it's fine, as I'm not sure if there are any side effects of modifying `data.ships`

> Should I make it so that we can easily toggle between bounding invalid coordinates to valid equivalents vs deleting AIS boats with invalid coordinates?

I think just deleting is fine because the invalid coordinates we saw was `(91, 181)`, which I highly doubt is a real boat. Plus we only want to look at boats that are close to us

File: python/Sailbot.py [hunk 1]
```diff
@@ -176,6 +176,17 @@ def globalWindCallback(self, data):
         self.globalWindLastUpdate = time.time()
 
     def AISCallback(self, data):
+        filteredShips = []
+
+        for ship in data.ships:
+            if utils.checkLatLonIsValid(latlon(ship.lat, ship.lon)):
```
[the remark this answers is not in the document]
Comment (jamenkaye) [reply python/Sailbot.py:182]: Makes sense, I'm addressing it in the next commit.

File: python/Sailbot.py [hunk 2]
```diff
@@ -176,6 +176,17 @@ def globalWindCallback(self, data):
...
+                filteredShips.append(ship)
+
+            else:
+                rospy.logwarn_throttle(5, "Invalid co-ordinates for "
```
[the remark this answers is not in the document]
Comment (patrick-5546) [reply python/Sailbot.py]: I addressed this by taking the throttled warning out of the for-loop.

Current error message:
image

**Review by patrick-5546 — APPROVED** [review]

File: python/Sailbot.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jamenkaye) [reply python/Sailbot.py]: Ok ya I think this is best, for some reason I thought the throttling wouldn't apply if the message was different each time

**Pull request merged**